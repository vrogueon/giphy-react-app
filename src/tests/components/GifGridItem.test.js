import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme/build';
import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas en <GifGridItem/>', () => {
    
    const title = 'Ejemplo titulo';
    const url = 'https://localhost/algo.jpg';
    
    let wrapper = shallow(<GifGridItem title={ title } url={ url }/>)

    test('debe de mostrar el <GifGridItem/> Correctamente ', () => {
        expect( wrapper ).toMatchSnapshot();
    });

    test('debe de tener un parrafo con texto', () => {
        const p = wrapper.find( 'p' );
        expect(p.text().trim()).toBe( title );
    });
    
    test('debe de tener la imagen igual a la url y alt de los props', () => {
       const img = wrapper.find( 'img' );
       
       expect( img.prop( 'src' ) ).toBe( url );
       expect( img.prop( 'alt' ) ).toBe( title );
    });

    test('debe de tener animate__fadeIn', () => {
        const div = wrapper.find( 'div' );
        // expect(div.hasClass('animate__fadeIn')).toBe(true);
        const className = div.prop( 'className' );
        expect(className.includes( 'animate__fadeIn' )).toBe( true );
    })
    
    
    
});
