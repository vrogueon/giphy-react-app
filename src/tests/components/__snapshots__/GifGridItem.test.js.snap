// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem/> debe de mostrar el <GifGridItem/> Correctamente  1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Ejemplo titulo"
    src="https://localhost/algo.jpg"
  />
  <p>
    Ejemplo titulo
  </p>
</div>
`;
